ui-v2/app/helpers/format-time.js

```diff
@@ -0,0 +1,29 @@
+import { helper } from '@ember/component/helper';
+
+export default helper(function formatTime([params], hash) {
+  let day, hour, minute, seconds;
+  seconds = Math.floor(params / 1000);
+  minute = Math.floor(seconds / 60);
+  seconds = seconds % 60;
+  hour = Math.floor(minute / 60);
+  minute = minute % 60;
+  day = Math.floor(hour / 24);
+  hour = hour % 24;
+  const time = {
+    day: day,
+    hour: hour,
+    minute: minute,
+    seconds: seconds,
+  };
+
+  switch (true) {
+    case time.day != 0:
+      return time.day + 'd';
+    case time.hour != 0:
+      return time.hour + 'h';
+    case time.minute != 0:
+      return time.minute + 'm';
+    default:
+      return time.seconds + 's';
+  }
+});
```

Comment:
This (the where you use it) is a great improvement on what was there. Love that you moved this all into a component and improved it at the same time 👏 , it should have been like this from the start.

Little note here, I don't think we should just yet, but there are various date/time helper libraries out there I think we could consider when the time is right (honestly didn't mean that pun!). I've also been looking at a few libs to replace some of our string and array functions that we wrote a long time ago, just a small note for future really.

Ah just spotted one thing, no big deal at all, but we could go with strict equality above here `!==` vs `!=` and template strings, thats a super nitpick tho, so feel free to leave as is.